app/src/main/java/net/gini/android/gvlexample/GVLExampleApp.java

```diff
@@ -0,0 +1,42 @@
+package net.gini.android.gvlexample;
+
+import android.app.Application;
+
+import net.gini.android.Gini;
+import net.gini.android.SdkBuilder;
+import net.gini.android.gvlexample.gini.SingleDocumentAnalyzer;
+
+/**
+ * Created by Alpar Szotyori on 27.11.2017.
+ *
+ * Copyright (c) 2017 Gini GmbH.
+ */
+
+public class GVLExampleApp extends Application {
+
+    private Gini mGiniApi;
+    private SingleDocumentAnalyzer mSingleDocumentAnalyzer;
+
+    public SingleDocumentAnalyzer getSingleDocumentAnalyzer() {
+        if (mSingleDocumentAnalyzer == null) {
+            mSingleDocumentAnalyzer = new SingleDocumentAnalyzer(getGiniApi());
+        }
+        return mSingleDocumentAnalyzer;
+    }
+
+    public Gini getGiniApi() {
+        if (mGiniApi == null) {
+            createGiniApi();
+        }
+        return mGiniApi;
+    }
+
+    private void createGiniApi() {
+        final String clientId = this.getString(R.string.gini_api_client_id);
+        final String clientSecret = this.getString(R.string.gini_api_client_secret);
+        SdkBuilder builder = new SdkBuilder(this, clientId,
+                clientSecret,
+                "gvlexample.net");
```

Comment:
Why did you extracted the _id_ and _secret_ to your resources but not the domain?

Reply:
We need to inject the client credentials at build time. For this I inject string resources using the android gradle plugin. The values are taken either from project properties or a `local.properties` file, which is not tracked.

The domain is static.